### PR TITLE
WIP: Use Kubernetes latest release 1.20

### PR DESCRIPTION
**Pull request by fmuyassarov**
Install k8s 1.20 binaries on the target node image. Use CRI-O as container runtime for kubelet instead of Docker.

### CODE REVIEW

**Review by namnx228 — DISMISSED** [review]

File: ci/scripts/image_scripts/provision_node_image.sh [hunk 1]
```diff
@@ -4,9 +4,15 @@ set -uex
...
 
-export KUBERNETES_VERSION=${KUBERNETES_VERSION:-"v1.19.3"}
+export KUBERNETES_VERSION=${KUBERNETES_VERSION:-"v1.20.0"}
 export KUBERNETES_BINARIES_VERSION="${KUBERNETES_BINARIES_VERSION:-${KUBERNETES_VERSION}}"
 export KUBERNETES_BINARIES_CONFIG_VERSION=${KUBERNETES_BINARIES_CONFIG_VERSION:-"v0.2.7"}
+export CRIOCTL_VERSION=${CRIOCTL_VERSION:-"v1.20.0"}
+# CRI-O required vars
+OS=xUbuntu_20.04
+# This is Kubernetes version
+VERSION=1.20
```
Comment (kashifest) [inline ci/scripts/image_scripts/provision_node_image.sh:14]: Do we need default values for these or should this be hardcoded ?

File: ci/scripts/image_scripts/provision_node_image.sh [hunk 2]
```diff
@@ -36,8 +42,48 @@ sudo apt install -y keepalived && sudo systemctl stop keepalived
...
+sudo apt-get install cri-o cri-o-runc -y
+
+# Start CRI-O
+sudo systemctl daemon-reload
+sudo systemctl start crio
+
+# Download crictl
+curl -LO https://github.com/kubernetes-sigs/cri-tools/releases/download/${CRIOCTL_VERSION}/crictl-${CRIOCTL_VERSION}-linux-amd64.tar.gz
+sudo tar -C /usr/local/bin -xzf crictl-${CRIOCTL_VERSION}-linux-amd64.tar.gz
+
```
Comment (kashifest) [inline ci/scripts/image_scripts/provision_node_image.sh:86]: can we take this in a separate script like it was for docker ?

File: ci/scripts/image_scripts/provision_node_image_centos.sh [hunk 3]
```diff
@@ -3,9 +3,47 @@
...
+sudo curl -L -o /etc/yum.repos.d/devel:kubic:libcontainers:stable.repo https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/devel:kubic:libcontainers:stable.repo
+sudo curl -L -o /etc/yum.repos.d/devel:kubic:libcontainers:stable:cri-o:$VERSION.repo https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:$VERSION/$OS/devel:kubic:libcontainers:stable:cri-o:$VERSION.repo
+sudo yum install cri-o -y
+
+sudo systemctl daemon-reload
+sudo systemctl start crio
+
+# Download crictl
+curl -LO https://github.com/kubernetes-sigs/cri-tools/releases/download/${CRIOCTL_VERSION}/crictl-${CRIOCTL_VERSION}-linux-amd64.tar.gz
+sudo tar -C /usr/local/bin -xzf crictl-${CRIOCTL_VERSION}-linux-amd64.tar.gz
```
Comment (kashifest) [inline ci/scripts/image_scripts/provision_node_image_centos.sh:46]: Same comment as above. May be a separate script

File: ci/scripts/image_scripts/upload_node_image_rt.sh [hunk 4]
```diff
@@ -8,14 +8,15 @@
 #   2. uploads it to artifactory
 # 
 # Usage:
-#   ./upload_node_image_rt.sh <IMAGE_NAME> <IMAGE_SIZE>
+#   ./upload_node_image_rt.sh <IMAGE_NAME>
 #   ./upload_node_image_rt.sh CENTOS_8.2_NODE_IMAGE_K8S_v1.18.8
+# Don't include image the format, only the name. CENTOS_8.2_NODE_IMAGE_K8S_v1.18.8 instead of CENTOS_8.2_NODE_IMAGE_K8S_v1.18.8.qcow2
```
Comment (kashifest) [inline ci/scripts/image_scripts/upload_node_image_rt.sh:13]: # Don't include the image format

**Review by kashifest — CHANGES_REQUESTED** [review]
Few small cosmetic changes  needed, otherwise looks good, thanks for taking care of this.

File: ci/scripts/image_scripts/provision_node_image.sh (hunk 1 above)
Comment (fmuyassarov) [reply ci/scripts/image_scripts/provision_node_image.sh:14]: Not sure if there will be benefit of defaulting.

File: ci/scripts/image_scripts/provision_node_image.sh (hunk 2 above)
Comment (fmuyassarov) [reply ci/scripts/image_scripts/provision_node_image.sh:42]: Yes sure. Done

File: ci/scripts/image_scripts/provision_node_image.sh (hunk 1 above)
Comment (fmuyassarov) [reply ci/scripts/image_scripts/provision_node_image.sh]: Fixed it.

**Comment by fmuyassarov** [comment]
@kashifest Thanks for the review. Addressed your comments. Please take a look.

**Review by kashifest — APPROVED** [review]

**Pull request merged**